griptape/drivers/vector/pgvector_vector_store_driver.py

```diff
@@ -0,0 +1,159 @@
+import uuid
+from typing import Optional
+from attr import define, field, Factory
+from griptape.drivers import BaseVectorStoreDriver
+from sqlalchemy.engine import Engine
+from sqlalchemy import create_engine, Column, String, JSON
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Session
+from pgvector.sqlalchemy import Vector
+
+
+@define
+class PgVectorVectorStoreDriver(BaseVectorStoreDriver):
+    """
+    A vector store driver to Postgres using the PGVector extension.
+
+    Attributes:
+        connection_string: The connection string for the Postgres database instance.
+        create_engine_params: Additional configuration params passed when creating the database connection.
+        engine: An optional sqlalchemy Postgres engine to use.
+    """
+
+    connection_string: Optional[str] = field(default=None, kw_only=True)
+    create_engine_params: Optional[dict] = field(factory=dict, kw_only=True)
+    engine: Optional[Engine] = field(
+        default=Factory(
+            lambda self: create_engine(self.connection_string, **self.create_engine_params), takes_self=True
+        )
+    )
+
+    class VectorModel(declarative_base()):
+        __tablename__ = "griptape_vectors"
+
+        id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, unique=True, nullable=False)
+        vector = Column(Vector())
+        namespace = Column(String)
+        meta = Column(JSON)
+
+    def install_postgres_extensions(self) -> None:
```

Comment:
Several of our prompt drivers implement functionality to "initialize" themselves. i.e. set up indexes/tables/etc. Should we move this to the base class so there is a standard interface?

Reply:
@vasinov what're your thoughts on this?

Reply:
I originally decided against it because, at the time, it seemed like different vector store initialization processes were too unique for an abstract method and it felt like more of a bonus thing that _some_ drivers would provide.

We should re-evaluate that assumption though. It might be nice to have a standard method with `**kwargs` in it that child classes optionally implement.

Reply:
I've created an initialization method that we can propagate to other vector store classes if we'd like. We can discuss further.